pkg/orchestrator/gitHubActions.go

```diff
@@ -1,88 +1,227 @@
 	return "n/a"
 }
 
 func (g *GitHubActionsConfigProvider) OrchestratorType() string {
-	return "GitHubActions"
+	return "GitHub Actions"
 }
 
 func (g *GitHubActionsConfigProvider) GetBuildStatus() string {
-	log.Entry().Infof("GetBuildStatus() for GitHub Actions not yet implemented.")
-	return "FAILURE"
+	// By default, we will assume it's a success
+	// On error it would be handled by the action itself
+	return "SUCCESS"
 }
 
 func (g *GitHubActionsConfigProvider) GetLog() ([]byte, error) {
-	log.Entry().Infof("GetLog() for GitHub Actions not yet implemented.")
-	return []byte{}, nil
+	resp, err := g.client.GetRequest(
+		fmt.Sprintf(
+			"%s/runs/%s/jobs", getActionsURL(), getEnv("GITHUB_RUN_ID", ""),
+		),
+		map[string][]string{
+			"Accept":        {"application/vnd.github+json"},
+			"Authorization": {"Bearer $GITHUB_TOKEN"},
+		}, nil)
+	if err != nil {
+		return nil, fmt.Errorf("can't get API data: %w", err)
+
+	}
+	var ids struct {
+		Jobs []struct {
+			Id string `json:"id"`
+		} `json:"jobs"`
+	}
+	err = piperHttp.ParseHTTPResponseBodyJSON(resp, &ids)
+	if err != nil {
+		return nil, fmt.Errorf("can't parse JSON data: %w", err)
+	}
+	ids = struct {
+		Jobs []struct {
+			Id string `json:"id"`
+		} `json:"jobs"`
+		// we cant get the log for the last(current) job
+	}{Jobs: ids.Jobs[:len(ids.Jobs)-1]}
+	if len(ids.Jobs) == 0 {
+		return nil, fmt.Errorf("can't get the log form the last(current) job")
+	}
+	logs := struct {
+		b [][]byte
+		sync.Mutex
+	}{
+		b: make([][]byte, len(ids.Jobs)),
+	}
+	ctx := context.TODO()
+	sem := semaphore.NewWeighted(10)
+	wg := errgroup.Group{}
+	for i := range ids.Jobs {
+		if err := sem.Acquire(ctx, 1); err != nil {
+			return nil, fmt.Errorf("failed to acquire semaphore: %w", err)
+		}
+		j := i
+		wg.Go(func() error {
+			defer sem.Release(1)
+			resp, err := g.client.GetRequest(
+				fmt.Sprintf(
+					"%s/jobs/%s/logs", getActionsURL(), ids.Jobs[j].Id,
+				),
+				map[string][]string{
+					"Accept":        {"application/vnd.github+json"},
+					"Authorization": {"Bearer $GITHUB_TOKEN"},
+				}, nil)
+			if err != nil {
+				return fmt.Errorf("can't get API data: %w", err)
+
+			}
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return fmt.Errorf("can't read response body: %w", err)
+			}
+			logs.Lock()
+			defer logs.Unlock()
+			logs.b[j] = append([]byte{}, body...)
+			return nil
+		})
+	}
+	if err = wg.Wait(); err != nil {
+		return nil, fmt.Errorf("recieving log error: %w", err)
+	}
+	return bytes.Join(logs.b, []byte("")), nil
 }
 
 func (g *GitHubActionsConfigProvider) GetBuildID() string {
-	log.Entry().Infof("GetBuildID() for GitHub Actions not yet implemented.")
-	return "n/a"
+	return getEnv("GITHUB_RUN_ID", "n/a")
 }
 
 func (g *GitHubActionsConfigProvider) GetChangeSet() []ChangeSet {
-	log.Entry().Warn("GetChangeSet for GitHubActions not yet implemented")
-	return []ChangeSet{}
+	return []ChangeSet{
```

Comment:
What is the ChangeSet being returned here? Info: This was introduced for the DORA use case and could be removed from all orchestrator types in future.